### PR TITLE
Ports + LibC: port the c-ares library, define the missing IPv6 addr macros needed in netinet/in.h

### PR DESCRIPTION
**Pull request by autumn-traveller**
This PR ports the c-ares library, a library for making asynchronous DNS requests. It was very close to cross-compiling on its own without any changes but there were a couple of IPv6 related macros missing from LibC in netinent/in.h. I defined these as best I could based off of the relevant RFC specifications.

Please let me know if there is anything I can improve :)

### CODE REVIEW

File: Ports/c-ares/package.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='c-ares'
+version='1.19.0'
+files="https://c-ares.org/download/c-ares-${version}.tar.gz c-ares-${version}.tar.gz
+https://c-ares.org/download/c-ares-${version}.tar.gz.asc c-ares-${version}.tar.gz.asc"
+auth_type='sig'
+auth_import_key="27EDEAF22F3ABCEB50DB9A125CC908FDB71E12C2"
+auth_opts=("c-ares-${version}.tar.gz.asc" "c-ares-${version}.tar.gz")
```
Comment (timschumi) [inline Ports/c-ares/package.sh:8]: Please validate files via sha256sums instead.

File: Ports/c-ares/package.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+useconfigure='true'
+configopts=("-DCMAKE_TOOLCHAIN_FILE=${SERENITY_BUILD_DIR}/CMakeToolchain.txt")
+
+configure() {
+	mkdir -p c-ares-build
+	cmake -G Ninja \
+	"${configopts[@]}" \
+	-S "c-ares-${version}" \
+	-B c-ares-build
```
Comment (timschumi) [inline Ports/c-ares/package.sh:17]: Please indent those lines, to make the continuation more obvious.

File: Ports/c-ares/package.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+build() {
+	ninja -C c-ares-build -j`nproc`
```
Comment (timschumi) [inline Ports/c-ares/package.sh:21]: We should avoid using `nproc` directly, some users may want to use a smaller number of build jobs. We already define a `MAKEJOBS` variable, which one should use for ports scripts instead.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/c-ares/package.sh (hunk 1 above)
Comment (autumn-traveller) [reply Ports/c-ares/package.sh]: Sure no problem

File: Ports/c-ares/package.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+files="https://c-ares.org/download/c-ares-${version}.tar.gz c-ares-${version}.tar.gz bfceba37e23fd531293829002cac0401ef49a6dc55923f7f92236585b7ad1dd3"
+auth_type='sha256'
+useconfigure='true'
+configopts=("-DCMAKE_TOOLCHAIN_FILE=${SERENITY_BUILD_DIR}/CMakeToolchain.txt")
+
+configure() {
+	mkdir -p c-ares-build
+	cmake -G Ninja \
+		"${configopts[@]}" \
+		-S "c-ares-${version}" \
```
Comment (ADKaster) [inline Ports/c-ares/package.sh:13]: Tim can correct me if I'm misremembering, but this should be handled by `workdir`, yes?

`run cmake <blah>` chould `cd` to the workdir and handle the -S part at least.

Also there's tabs in this file, pls convert to 4 spaces.

File: Ports/c-ares/package.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='c-ares'
+version='1.19.0'
+files="https://c-ares.org/download/c-ares-${version}.tar.gz c-ares-${version}.tar.gz bfceba37e23fd531293829002cac0401ef49a6dc55923f7f92236585b7ad1dd3"
+auth_type='sha256'
+useconfigure='true'
+configopts=("-DCMAKE_TOOLCHAIN_FILE=${SERENITY_BUILD_DIR}/CMakeToolchain.txt")
```
Comment (ADKaster) [inline Ports/c-ares/package.sh:7]: While it's nice that the new IPv6 defintions make ares happier, we should pass whatever  "disable IPv6" configure option exists for the library in configopts here. Trying to use the mostly-unimplemented IPv6 support in the serenity kernel will only lead to pain. Patches welcome to make IPv6 actually work :)

File: Ports/c-ares/package.sh (hunk 4 above)
Comment (autumn-traveller) [reply Ports/c-ares/package.sh]: Sorry about the tabs! Fixed that in the latest push. I have switched line 13 to use $workdir instead

File: Ports/c-ares/package.sh (hunk 5 above)
Comment (autumn-traveller) [reply Ports/c-ares/package.sh:7]: So unfortunately I dont think there is a build option to disable IPv6. I looked through bits of their source code and especially the paths where the IPv6 definitions we had to add are used, but there are no points where anything leading to the macros I had to define can be optionally build or not. Just to be safe I also tried building with disable-ipv6 and similar options.

Comment (ADKaster) [reply Ports/c-ares/package.sh:7]: Ah, that's unfortunate. I confirmed there's no such option by looking at the CMakeLists.txt upstream.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**